README.md

```diff
@@ -22,77 +22,96 @@ create a plethora of workload characteristics that rely on a common storage mode
 
-ChronoLog uses HCL internally. It is added to this repository as a submodule. Thus, you need to clone the submodules as
-well. You can do it using `git clone --recursive git@github.com:scs-lab/ChronoLog.git` to clone ChronoLog. Or you can
-run `git submodule update --init --recursive` once in `ChronoLog` directory after you clone the repository
-without `--recursive`. For following pulls, you can update the submodule using command `git pull --recurse-submodules`.
+**Option 1: Clone directly with Submodules (Clone and automatically initializes and updates the submodules)**
 
-## Building
+```
+git clone --recursive https://github.com/grc-iit/ChronoLog.git
+```
+
+**Option 2: Initialize Submodules After Cloning**
+
+If you prefer or need to separate the cloning and submodule initialization into two steps, follow these instructions:
+
+```
+git clone https://github.com/grc-iit/ChronoLog.git
+git submodule update --init --recursive
+```
+
+**Future Updates with Submodules**
+
+For subsequent updates, ensure you also update the submodules alongside the main repository by using:
+
+```
+git pull --recurse-submodules
+```
 
-ChronoLog has a list of dependencies which can be solved by Spack packages. Thus, Spack needs to be installed and
-configured as the first step to build ChronoLog.
+## Installation and configuration
 
-### Install Spack
+### Prerequisites: Spack
 
-Spack can be checked out with `git clone https://github.com/spack/spack.git`. It is assumed that Spack is stored
-at `~/Spack` for the following step. Spack needs to activated by
-running `source ~/Spack/spack/share/spack/setup-env.sh`.
+ChronoLog requires various packages managed by Spack. To ensure compatibility and stability, we recommend using Spack version [`v0.21.2 (2024-03-01)`](https://github.com/spack/spack/releases/tag/v0.21.2). Follow the steps below to install and configure Spack:
 
-### Install ChronoLog dependencies
+```
+git clone --branch v0.21.2 https://github.com/spack/spack.git
+source /path-to-where-spack-was-cloned/spack/share/spack/setup-env.sh
+```
 
-Currently, most of the dependencies are listed in `spack.yaml` and can be installed via Spack. `gcc` and `g++` will be
-needed to build ChronoLog.
+### Installing Dependencies
 
-A Spack environment needs to be created and activated using the following commands. When the environment is activated, a
-shell prompt `[ChronoLog]` will pop up.
+Currently, most of the dependencies are listed in `spack.yaml` and can be installed via Spack.`gcc` and `g++` will be needed to build ChronoLog.
 
+A Spack environment needs to be created and activated using the following commands. When the environment is activated, a shell prompt `[ChronoLog]` will pop up.
 ```
 cd ChronoLog
 git switch develop
 spack env create -d .
```

Comment:
make sure env create -d . is necessary. I think we can skip this one when we already have a `spack.yaml`.